In this notebook i'm going to get other model using random forest.

## 1º I get the model again

#### Import the libraries

In [1]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.ensemble import RandomForestRegressor
import pandas as pd

#### Load data

In [2]:
melbourne_file_path = '/Users/alvaro/Kaggle/Machine Learning/2 - Explore the data/melb_data.csv'
melbourne_data = pd.read_csv(melbourne_file_path)

#### Filter rows with missing values

In [3]:
melbourne_data = melbourne_data.dropna(axis=0)

#### Choose target and features

In [4]:
y = melbourne_data.Price
melbourne_features = ['Rooms', 'Bathroom', 'Landsize','BuildingArea','YearBuilt','Lattitude', 'Longtitude']
X = melbourne_data[melbourne_features]

#### Split data into training and validation data, for both features and target

In [5]:
train_X, test_X, train_y, test_y = train_test_split(X, y, random_state = 0)

#### Predict

In [6]:
forest_model = RandomForestRegressor(random_state=1)
forest_model.fit(train_X, train_y)
melb_preds = forest_model.predict(test_X)

#### Get MAE

In [7]:
print(mean_absolute_error(test_y, melb_preds))

202806.6128254788


## The MAE reduce from 243176.44 dollars using decission tree with 500 leaves to $202806.61 using random forest model, It's a 16,6% minus